src/validation.cairo

```diff
@@ -50,7 +50,13 @@ impl TransactionValidatorImpl of TransactionValidator {
 }
 
 fn block_hash(self: @ChainState, block: @Block, merkle_root: u256) -> Result<u256, ByteArray> {
-    // TODO: implement
+    let header = block.header;
+
+    let mut header_data = ArrayTrait::<u32>::new();
+    header_data.append(*header.version);
+    header_data.append(*header.time);
```

Comment:
Don't forget about `previous_block` and `merkle_root`. `previous_block` is in `self.best_block_hash`.

Reply:
I think we will change how hashes are represented from u256 to [u32; 8], so adding them to `header_data` will be easier.

Reply:
In order not to wait for above changes you can modify function signature to:
```
block_hash(version: u32, previous_block: [u32; 8], merkle_root: [u32; 8], time: u32, bits:u32, nonce:32) -> [u32; 8]
```

Reply:
Got it. But now how can I hash previous_block and merkle_root?